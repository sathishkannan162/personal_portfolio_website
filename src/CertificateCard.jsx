import ImageWithPlaceholder from './ImageWithPlaceholder';
import { Box, Image, Text, Flex, Spacer, Heading } from '@chakra-ui/react';
import certificateImage1 from './assets/certificates/certificate_1.png';
import { motion } from 'framer-motion';
import { useState } from 'react';
export default function CertificateCard() {
  const imageUrl = certificateImage1;
  const [hover, setHover] = useState(false);

  const handleHover = () => {
    setHover(!hover);
  };

  return (
    <Flex>
      <motion.div onHoverStart={handleHover} onHoverEnd={handleHover}>
        <Box
          maxW="500px"
          maxH="500px"
          minW="300px"
          aspectRatio={2.6 / 2}
          backgroundImage={imageUrl}
          backgroundSize="cover"
        >
          {/* <ImageWithPlaceholder src={imageUrl} /> */}
          {/* <Flex flexDir={'column-reverse'}> */}
          <Box h={'70%'} />
          <Box opacity={hover ? 0.9 : 0} backgroundColor={'white'} h="30%">
            <Heading color={'black'} lineHeight={7}>
              <Text as="h3" fontWeight={500} fontSize="md" textAlign={'center'}>
                Responsive Web Design Certification
              </Text>
              <Text textAlign={'center'} fontWeight={400} fontSize={'sm'}>
                FreeCodeCamp
              </Text>
            </Heading>
          </Box>
          {/* </Flex> */}
        </Box>
      </motion.div>
    </Flex>
  );
}
