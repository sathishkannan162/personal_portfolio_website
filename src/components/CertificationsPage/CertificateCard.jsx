import { Box, Text, Heading, Link, Image } from '@chakra-ui/react';
import { motion } from 'framer-motion';
import { useState } from 'react';
import placeholder_square from '../../assets/placeholder_square.webp';

export default function CertificateCard({ certificate }) {
  const [hover, setHover] = useState(false);

  const handleHover = () => {
    setHover(!hover);
  };

  return (
    <Box data-test="certificate-card">
      <Heading fontSize={'xl'} py={4} px={1}>
        {certificate.name}
      </Heading>
      <motion.div
        whileHover={{ scale: 1.1 }}
        initial={{ opacity: 0, translateY: '20px' }}
        whileInView={{ opacity: 1, translateY: 0 }}
        viewport={{ once: true }}
        onHoverStart={handleHover}
        onHoverEnd={handleHover}
        whileTap={{ scale: 0.9 }}
        transition={{ duration: 0.2 }}
        data-test="certificate-div"
      >
        <Link href={certificate.certificateUrl} target="_blank">
          <Box
            maxW="500px"
            maxH="500px"
            minW={{ base: '150px', sm: '250px', md: '350px' }}
            aspectRatio={1.5 / 1}
            backgroundImage={certificate.imageUrl}
            backgroundSize="cover"
          >
            <Box h={'80%'} w="100%">
              <Image src={placeholder_square} opacity={0} />
            </Box>
            <motion.div whileHover={{ scale: 1.1 }}>
              <Box
                opacity={{ base: 1, sm: hover ? 0 : 0.9 }}
                backgroundColor={'gray.200'}
                h="20%"
                outline="2px solid"
                mx="auto"
              >
                <Heading color={'black'} lineHeight={7}>
                  <Text fontWeight={800} fontSize={'lg'} textAlign={'center'}>
                    {certificate.name}
                  </Text>
                  <Text textAlign={'center'} fontWeight={600} fontSize={'sm'}>
                    {certificate.provider}
                  </Text>
                  <Text textAlign={'center'} fontWeight={600} fontSize={'xs'}>
                    Click Card to View
                  </Text>
                </Heading>
              </Box>
            </motion.div>
          </Box>
        </Link>
      </motion.div>
    </Box>
  );
}
