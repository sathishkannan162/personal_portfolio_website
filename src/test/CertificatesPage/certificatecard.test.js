import React from 'react';
import { render, screen } from '../test-utils'
import CertificateCard from '../../components/CertificationsPage/CertificateCard.jsx';

// Framer motion requires browsser
// Mock the framer-motion package
// jest.mock('framer-motion', () => ({
//   motion: {
//     div: ({ children }) => <div>{children}</div>, 
//   },
// }));

const mockCertificate = {
  name: 'Certificate Name',
  certificateUrl: 'https://example.com',
  imageUrl: 'https://example.com/image.jpg',
  provider: 'Certificate Provider',
};

describe('CertificateCard', () => {
  test('renders certificate card correctly', () => {
    class IntersectionObserver {
      constructor(callback, options) {
        this.callback = callback;
        this.options = options;
      }
      observe(target) {
        this.target = target;
        this.callback([
          {
            isIntersecting: true,
            target: this.target,
          },
        ]);
      }

      unobserve() { }
    }

    window.IntersectionObserver = IntersectionObserver;

    render(<CertificateCard certificate={mockCertificate} />);

    // Verify that the certificate name is rendered
    expect(screen.queryAllByText(mockCertificate.name)).toHaveLength(2)
    expect(screen.queryAllByText(mockCertificate.name)[0]).toBeInTheDocument();
    expect(screen.queryAllByText(mockCertificate.name)[1]).toBeInTheDocument();

    // Verify that the certificate link is rendered with the correct URL
    const linkElement = screen.getByRole('link')
    expect(linkElement).toHaveAttribute('href', mockCertificate.certificateUrl);
    expect(linkElement).toHaveAttribute('target', '_blank');

    // Verify that the certificate image placeholder is rendered
    const imageElement = screen.getByAltText('placeholder image');
    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveStyle({ opacity: '0' });
  });
  test('captures and compares component snapshot', () => {
    const { asFragment } = render(<CertificateCard certificate={mockCertificate} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
