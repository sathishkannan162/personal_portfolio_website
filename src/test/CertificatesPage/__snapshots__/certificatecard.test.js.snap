// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificateCard captures and compares component snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-0"
    data-test="certificate-card"
  >
    <h2
      class="chakra-heading css-sj68pu"
    >
      Certificate Name
    </h2>
    <div
      data-test="certificate-div"
      style="opacity: 0; transform: translateY(20px) translateZ(0);"
      tabindex="0"
    >
      <a
        class="chakra-link css-spn4bz"
        href="https://example.com"
        target="_blank"
      >
        <div
          class="css-1yj8dtn"
        >
          <div
            class="css-88irkg"
          >
            <img
              alt="placeholder image"
              class="chakra-image css-q7lffx"
              src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
            />
          </div>
          <div>
            <div
              class="css-iw9ipp"
            >
              <h2
                class="chakra-heading css-c7nxzi"
              >
                <p
                  class="chakra-text css-1oj1yll"
                >
                  Certificate Name
                </p>
                <p
                  class="chakra-text css-1rhvqo5"
                >
                  Certificate Provider
                </p>
                <p
                  class="chakra-text css-5t65gu"
                >
                  Click Card to View
                </p>
              </h2>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
